packages/core/src/workers/generations/requestGeneration.ts

```diff
@@ -0,0 +1,446 @@
+  const generations = new Generations(prisma, logger);
+  const safetyViolations = new SafetyViolations(prisma, logger);
+
+  logger.info("Running step: Check generation exists");
+  const generationRecord = await generations.byId(generationId);
+
+  if (!generationRecord) {
+    throw new NonRetriableError("Generation does not exist");
+  }
+
+  if (generationRecord.status !== GenerationStatus.REQUESTED) {
+    throw new NonRetriableError("Generation has already been processed");
+  }
+
+  await generations.setStatus(generationRecord.id, GenerationStatus.PENDING);
+
+  baseLogger.info("Running step: Lookup prompt");
+  const prompt = await prompts.get(promptId, appId);
+
+  if (!prompt) {
+    throw new NonRetriableError("Prompt does not exist");
+  }
+
+  let promptBody: string;
+
+  baseLogger.info("Running step: Format prompt template");
+
+  try {
+    promptBody = await prompts.formatPrompt(prompt.template, promptInputs);
+
+    const promptInputsHash = generations.generatePromptInputsHash(promptInputs);
+    await generations.update(generationId, {
+      promptText: promptBody,
+      promptInputs,
+      promptInputsHash,
+    });
+  } catch (err) {
+    logger.error(
+      err,
+      "Error formatting or saving prompt template: %s",
+      err instanceof Error ? err.message : err,
+    );
+
+    throw new NonRetriableError("Error formatting or saving prompt template", {
+      cause: err,
+    });
+  }
+
+  /**
+   * --------------------- Begin moderation ---------------------
+   */
+  if (moderationConfig.MODERATION_ENABLED === true) {
+    await generations.setStatus(
+      generationRecord.id,
+      GenerationStatus.MODERATING,
+    );
+
+    /**
+     * These are quiz-specific keys which we should ignore, as they're
+     * almost certainly going to fail checks (e.g. numbers)
+     * @TODO: Come up with a more robust solution for this
+     */
+    const ignoredInputKeys = [
+      "subject",
+      "ageRange",
+      "distractorToRegenerate",
+      "numberOfCorrectAnswers",
+      "numberOfDistractors",
+      "knowledge",
+      "transcript",
+      "fact",
+      "sessionId",
+    ];
+
+    const userInput = Object.entries(promptInputs)
+      .filter(([key, value]) => !!value && !ignoredInputKeys.includes(key))
+      .map(([, value]) => JSON.stringify(value));
+
+    logger.info("Running step: Detect languages");
+    const [englishScores, isEnglish] = checkEnglishLanguageScores(userInput);
+
+    if (!isEnglish && MODERATE_LANGUAGE) {
+      logger.info("Running step: Save flagged generation (not english)");
+      await generations.flagGeneration(
+        generationId,
+        "That looks like non-English text. We currently only support English.",
+        ModerationType.NOT_ENGLISH,
+        {
+          moderationMeta: { englishScores },
+        },
+      );
+      // Return early and don't process generation further
+      return;
+    }
+
+    //logger.info("Running step: Detect profanity");
```

Comment:
Think this was commented out but shall we delete this while we are at it?

Reply:
Done